solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -390,6 +392,34 @@ pub mod solana_ibc {
         .map_err(error::Error::TokenTransferError)
         .map_err(|err| error!((&err)))
     }
+
+    /// The body of this method is empty since it is called to
+    /// realloc the accounts only.  Anchor sets up the accounts
+    /// given in this call’s context before the body of the method is
+    /// executed.
+    pub fn realloc_accounts(
+        ctx: Context<ReallocAccounts>,
+        new_length: usize,
+    ) -> Result<()> {
+        let payer = &ctx.accounts.payer.to_account_info();
+        let account = &ctx.accounts.account.to_account_info();
+        let new_length = new_length.max(account.data_len());
```

Comment:
```suggestion
```
With the `if new_length > old_length {` this isn’t necessary.

Reply:
right

Reply:
Not sure if you’ve meant to resolve this without applying the suggestion?

Reply:
i mean i removed the `if new_length > old_length {` check instead

Reply:
ah, ok, I’ve missed that. I’d probably do the opposite but either works.